service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -267,7 +267,13 @@ public JsonElement getMessageProtocols() {
         for (MsgService service : msgServices) {
             JsonObject protocolObject = new JsonObject();
             protocolObject.addProperty("name", service.getServiceName());
-            protocolObject.addProperty("edition", service.getProtocolEdition());
+            try {
+                protocolObject.addProperty("edition", service.getProtocolEdition());
+            } catch (NoSuchMethodError | AbstractMethodError e) {
+                // Ignored intentionally in order to ensure compatibility with
+                // eiffel-remrem-semantics:2.2.0 and older.
+                log.warn("Failed due to older version of REMRem Semantics library, please use the library which is greater than 2.2.0 version.");
```

Comment:
I think the message is confusing as it mentions Semantics library. But how do you know that it's a semantics library? The problematics method is getProtocolEdition() and it's defined in eiffel-remrem-protocol-interface:2.1.2. So the message should be something like 'An old library, without implementation of MsgService.getProtocolEdition(), is used. Please, upgrade to a newer library implementing eiffel-remrem-protocol-interface:2.1.2 or higher.' The message should be logged as 'error' to indicate that the upgrade is really needed.

Reply:
Updated the error message